src/Build/Logging/BinaryLogger/BuildEventArgsReader.cs

```diff
@@ -0,0 +1,673 @@
+            }
+
+            if ((flags & BuildEventArgsFieldFlags.EndLineNumber) != 0)
+            {
+                result.EndLineNumber = ReadInt32();
+            }
+
+            if ((flags & BuildEventArgsFieldFlags.EndColumnNumber) != 0)
+            {
+                result.EndColumnNumber = ReadInt32();
+            }
+
+            return result;
+        }
+
+        private void SetCommonFields(BuildEventArgs buildEventArgs, BuildEventArgsFields fields)
+        {
+            buildEventArgs.BuildEventContext = fields.BuildEventContext;
+
+            if ((fields.Flags & BuildEventArgsFieldFlags.ThreadId) != 0)
+            {
+                buildEventArgsFieldThreadId.SetValue(buildEventArgs, fields.ThreadId);
+            }
+
+            if ((fields.Flags & BuildEventArgsFieldFlags.SenderName) != 0)
+            {
+                buildEventArgsFieldSenderName.SetValue(buildEventArgs, fields.SenderName);
+            }
+
+            if ((fields.Flags & BuildEventArgsFieldFlags.Timestamp) != 0)
+            {
+                buildEventArgsFieldTimestamp.SetValue(buildEventArgs, fields.Timestamp);
+            }
+        }
+
+        private ArrayList ReadPropertyList()
+        {
+            var properties = ReadStringDictionary();
+            if (properties == null)
+            {
+                return null;
+            }
+
+            var list = new ArrayList();
+            foreach (var property in properties)
+            {
+                var entry = new DictionaryEntry(property.Key, property.Value);
+                list.Add(entry);
+            }
+
+            return list;
+        }
+
+        private BuildEventContext ReadBuildEventContext()
+        {
+            int nodeId = ReadInt32();
+            int projectContextId = ReadInt32();
+            int targetId = ReadInt32();
+            int taskId = ReadInt32();
+            int submissionId = ReadInt32();
+            int projectInstanceId = ReadInt32();
+
+            var result = new BuildEventContext(
+                submissionId,
+                nodeId,
+                projectInstanceId,
+                projectContextId,
+                targetId,
+                taskId);
+            return result;
+        }
+
+        private Dictionary<string, string> ReadStringDictionary()
+        {
+            int count = ReadInt32();
+
+            if (count == 0)
+            {
+                return null;
+            }
+
+            Dictionary<string, string> result = new Dictionary<string, string>();
+            for (int i = 0; i < count; i++)
+            {
+                string key = ReadString();
+                string value = ReadString();
+                result[key] = value;
+            }
+
+            return result;
+        }
+
+        private class TaskItem : ITaskItem
+        {
+            public string ItemSpec { get; set; }
+            public Dictionary<string, string> Metadata { get; } = new Dictionary<string, string>();
```

Comment:
Should this be one of the fancy collections we normally use? Might be nice for consistency but I doubt we need any of the additional functionality.